Firestore/core/src/firebase/firestore/model/transform_operations.h

```diff
@@ -90,6 +98,55 @@ class ServerTimestampTransform : public TransformOperation {
   }
 };
 
+// TODO(mikelehen): ArrayTransform can only be used from Obj-C until we switch
+// to using FieldValue instead of FSTFieldValue.
+#if defined(__OBJC__)
+/**
+ * Transforms an array via a union or remove operation (for convenience, we use
+ * this class for both Type::ArrayUnion and Type::ArrayRemove).
+ */
+class ArrayTransform : public TransformOperation {
+ public:
+  ArrayTransform(const Type& type, const std::vector<FSTFieldValue*> elements)
+      : type_(type), elements_(elements) {
+  }
+
+  ~ArrayTransform() override {
+  }
+
+  Type type() const override {
+    return type_;
+  }
+
+  std::vector<FSTFieldValue*> elements() const {
+    return elements_;
+  }
+
+  bool operator==(const TransformOperation& other) const override {
+    if (other.type() == type()) {
+      auto arrayTransform = static_cast<const ArrayTransform&>(other);
+      return arrayTransform.elements() == elements();
+    }
+    return false;
+  }
+
+#if defined(__OBJC__)
```

Comment:
This #if is redundant but I left it in because it was in the other TransformOperation classes. I could remove it though...